main.go

```diff
@@ -76,15 +78,17 @@ func main() {
 }
 
 func createDirs(config Conf) error {
-	for _, arch := range config.SupportArch {
-		if _, err := os.Stat(config.ArchPath(arch)); err != nil {
-			if os.IsNotExist(err) {
-				log.Printf("Directory for %s does not exist, creating", arch)
-				if err := os.MkdirAll(config.ArchPath(arch), 0755); err != nil {
-					return fmt.Errorf("error creating directory for %s: %s", arch, err)
+	for _, distro := range config.DistroNames {
+		for _, arch := range config.SupportArch {
+			if _, err := os.Stat(config.ArchPath(distro, arch)); err != nil {
+				if os.IsNotExist(err) {
+					log.Printf("Directory for %s does not exist, creating", arch)
+					if err := os.MkdirAll(config.ArchPath(distro, arch), 0755); err != nil {
+						return fmt.Errorf("error creating directory for %s: %s", arch, err)
```

Comment:
These messages should include the distro. Something like:

`return fmt.Errorf("error creating directory for %s (%s): %s", distro, arch, err)`

`log.Printf("Directory for %s (%s) does not exist, creating", distro, arch)`